### DISCUSSION TITLE
Ability to drag and drop tabs into split view

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#892] [Ideas]
It's really tedious how you currently select tabs for split view. It would be better to be able to just drag and drop them in and out of split view, or just generally a better way to select and deselect tabs into and out of split view,

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/90_

**Comment by Iziram** [comment]
What could be interesting is a keyboard shortcut to show a "grid layout menu" (see following screen).
The default behaviour should be to place the current tab in the first grid space (from left to right, up to down) and let use drag other tabs in the other spaces (or a button to select a tab from a tab list)

![image](https://github.com/user-attachments/assets/ea5bbb5d-1886-40b8-af1c-8a6c84f9ec2a)

Bonus : Let user drag and drop tabs in the grid layout to swap between places.

**Reply by SomeDudeBrosky** [reply]
For people that do not use keybindings, a button in the header bar to bring up this menu would be great.

**Comment by zheshien0812** [comment]
Would be great if in Sidebar, we can drag and drop a tab onto another directly to create a default split view.

Something like this: https://x.com/Arc_eology/status/1751953338206449755.

**Reply by LaserKaspar** [reply]
That would be awesome! The current implementation of split tabs is not really intuitive and because of that I use it only when absolutely necessary...

**Reply by pledi** [reply]
This is really intuitive and easy to adapt for new users! I used it quiet a lot on arc. 

Another way would be to just drag a tab onto the screen displaying the fitting split based on the elements position

**Comment by cowboycodr** [comment]
+1 This is a must!

**Comment by Krazy303** [comment]
It is now implemented in zen so you should close this discussion i guess

**Comment by fidelix** [comment]
Missing still is the ability to drop one tab into another to form a split view.

**Reply by maxhoffmann** [reply]
Yes, dragging tabs onto each other should be supported as well. Also right-clicking a split view tab to unsplit is still missing

**Reply by fidelix** [reply]
> Yes, dragging tabs onto each other should be supported as well. Also right-clicking a split view tab to unsplit is still missing

I just tested and dragging tabs onto each other does nothing but switch tab places. I have the latest stable version.

**Reply by maxhoffmann** [reply]
I know. I agreed that this should be supported as well.

**Comment by illegalnumbers** [comment]
Is this still not supported?

**Reply by Krazy303** [reply]
It is now

**Reply by fidelix** [reply]
Yes. It just doesn't support arc-style mechanism where you drag one tab into the other, which is objectively superior to the way it currently works in Zen.

**Reply by mr-cheffy** [reply]
Thats also in the works, there's a PR for it

**Reply by tomihbk** [reply]
> Thats also in the works, there's a PR for it

Hey, do you the PR number ? I want to follow it. thanks

**Reply by sporocyst** [reply]
#12320
